src/lib/list.c

```diff
@@ -388,3 +388,50 @@ struct list *list_filter_common_elements(struct list *list1, struct list *list2,
 
 	return preserver;
 }
+
+struct list *list_str_deduplicate(struct list *list)
+{
+	list = list_sort(list, list_strcmp);
+	return list_deduplicate(list, list_strcmp, free);
+}
+
+void *list_remove(void *item_to_remove, struct list **list, comparison_fn_t comparison_fn, list_free_data_fn_t list_free_data_fn)
+{
+	struct list *iter = NULL;
+	void *item = NULL;
+	int comp;
+
+	*list = iter = list_head(*list);
+
+	for (; iter; iter = iter->next) {
+
+		item = iter->data;
+
+		comp = comparison_fn(item, item_to_remove);
+		if (comp == 0) {
+			if (*list == iter) {
+				/* iter should always point to the head of the list,
+				 * if that is the element being removed, we need to
+				 * point to the new head */
+				*list = list_free_item(iter, list_free_data_fn);
+			} else {
+				list_free_item(iter, list_free_data_fn);
+			}
+			/* if the data was freed, it will return NULL */
+			return item;
+		}
+	}
+
+	/* item not found */
+	return NULL;
+}
+
+void list_move_item(void *item_to_move, struct list **list1, struct list **list2, comparison_fn_t comparison_fn)
+{
+	void *item;
+
+	item = list_remove(item_to_move, list1, comparison_fn, NULL);
```

Comment:
I would put this into a loop to move all occurrences of item to the new list.

Reply:
I can do that but we are going to have a problem when the data is being freed, because then list_remove is going to return NULL which would be the same result we would get if the item was not found in the list.... I can implement this but I would have to remove the option of freeing the data, so it would only remove the data from the list. You would manually have to free it if you want to clean it up

Reply:
ooh wait, never mind haha my bad, I'll do this change

Reply:
done